ADAL/tests/unit/ADAuthenticationErrorTests.m

```diff
@@ -40,70 +40,88 @@ - (void)tearDown
+    ADAssertStringEquals(error.errorDetails, @"The argument 'parameter123456 %@' is invalid. Value:(null)");
 }
 
-- (void)testErrorFromArgumentNormal
+- (void)testErrorFromArgument_whenArgumentIsValidArgumentNameIsValidCorrelationIdNil_shouldReturnError
 {
-    NSString* parameter = @"parameter123456 %@";
-    NSString* parameterValue = @"value1245 %s@";
-    ADAuthenticationError* error = [ADAuthenticationError errorFromArgument:parameterValue argumentName:parameter correlationId:nil];
-    XCTAssertNotNil(error, "No error for valid prameter");
+    NSString *parameter = @"parameter123456 %@";
+    NSString *parameterValue = @"value1245 %s@";
     
-    [self adValidateForInvalidArgument:parameter error:error];
-    XCTAssertTrue([error.errorDetails containsString:parameterValue], "Value should be part of the text");
+    ADAuthenticationError *error = [ADAuthenticationError errorFromArgument:parameterValue argumentName:parameter correlationId:nil];
+    
+    XCTAssertNotNil(error);
+    ADAssertStringEquals(error.domain, ADAuthenticationErrorDomain);
+    XCTAssertNil(error.protocolCode);
+    ADAssertStringEquals(error.errorDetails, @"The argument 'parameter123456 %@' is invalid. Value:value1245 %s@");
 }
 
-- (void)testErrorFromOAuthError
+#pragma mark - errorFromAuthenticationError
+
+- (void)testErrorFromAuthenticationError_whenErrorOAuthHprotocolCodeValiderrorDetailsValidCorrelationIdNil_shouldReturnError
 {
-    NSString* details = @"Some details";
-    NSString* protocolCode = @"procol code";
-    ADAuthenticationError* error = [ADAuthenticationError errorFromAuthenticationError:AD_ERROR_SERVER_OAUTH protocolCode:protocolCode errorDetails:details correlationId:nil];
+    NSString *details = @"Some details";
+    NSString *protocolCode = @"procol code";
+    
+    ADAuthenticationError *error = [ADAuthenticationError errorFromAuthenticationError:AD_ERROR_SERVER_OAUTH protocolCode:protocolCode errorDetails:details correlationId:nil];
+    
     ADAssertStringEquals(error.domain, ADAuthenticationErrorDomain);
     ADAssertLongEquals(error.code, AD_ERROR_SERVER_OAUTH);
     ADAssertStringEquals(error.protocolCode, protocolCode);
     ADAssertStringEquals(error.errorDetails, details);
 }
 
-- (void)testDescription
+- (void)testErrorFromAuthenticationError_whenErrorValidprotocolCodeValiderrorDetailsValidCorrelationIdNil_shouldReturnErrorWithDescription
 {
-    NSString* details = @"Some details";
-    NSString* protocolCode = @"some-protocol-code";
-    ADAuthenticationError* error = [ADAuthenticationError errorFromAuthenticationError:42 protocolCode:protocolCode errorDetails:details correlationId:nil];
-    XCTAssertTrue([error.description containsString:details]);
-    XCTAssertTrue([error.description containsString:protocolCode]);
+    ADAuthenticationError *error = [ADAuthenticationError errorFromAuthenticationError:42 protocolCode:@"some-protocol-code" errorDetails:@"Some details" correlationId:nil];
```

Comment:
changed to this one: testErrorFromAuthenticationError_whenErrorValidCodeValidErrorDetailsValidCorrelationIdNil_shouldReturnErrorWithDescription Please let me know if you have a better name :)